Yoast/wordpress-seo
# Show content image for twitter

## Context


* The content image  fallback for twitter didn't work. I've reintroduced the code that is removed in #14354 and I removed the code from #14699. I also cherry-picked the commits from #13920

## Summary


This PR can be summarized in the following changelog entry:

* Improves the way we are getting the gallery images. Props to @stodorovic 

## Relevant technical choices:

* Reintroduced the code that is removed in #14354 
* Removed the code added in #14699
* Cherry picked commits from #13920
* Adds some `@covers` to make sure the coverage can be calculated correctly
* Removed two test files that are located identically at another location. 

## Test instructions

This PR can be tested by following these steps:
* Add some content with an image as post content
* Add some content with an image for the term the description.
* When Open Graph is enabled the Twitter image tag isn't visible.
* When Open Graph is diabled the Twitter image tag should be visible.

## UI changes

## Documentation

## Quality assurance


Fixes #14715
